sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/run/ArgumentsInputDialogImpl.kt

```diff
@@ -0,0 +1,47 @@
+package app.cash.sqldelight.intellij.run
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.layout.panel
+import javax.swing.JComponent
+
+internal interface ArgumentsInputDialog {
+  val result: List<SqlParameter>
+
+  fun showAndGet(): Boolean
+
+  interface Factory {
+    fun create(project: Project, parameters: List<SqlParameter>): ArgumentsInputDialog
+  }
+}
+
+internal class ArgumentsInputDialogImpl(
+  project: Project,
+  private val parameters: List<SqlParameter>
+) : DialogWrapper(project), ArgumentsInputDialog {
+
+  init {
+    init()
+  }
+
+  private val _result = mutableListOf<SqlParameter>()
+  override val result: List<SqlParameter> = _result
```

Comment:
super nit:
```suggestion
  override val result: List<SqlParameter> get() = _result
```
avoid redundant backing property